slave-base/contrib/bin/run-jnlp-client

```diff
@@ -8,7 +8,11 @@
 # The directory has to be writeable for the user that the container is running
 # under.
 
-#NOTE:  periodically check https://ce-gitlab.usersys.redhat.com/ce/jboss-dockerfiles/blob/develop/scripts/os-java-run/added/java-default-options for updates
+# NOTE:  periodically check https://ce-gitlab.usersys.redhat.com/ce/jboss-dockerfiles/blob/develop/scripts/os-java-run/added/java-default-options for updates
+
+MAX_RETRIES=5
```

Comment:
```bash

checkConnection() {
  MAX_RETRIES=5
  INTERVAL=1
  JENKINS_MASTER="http://localhost:8080"

  i=1
  echo "INFO: $(date +%T)[$i]: Checking connection to $JENKINS_MASTER"
  until curl -L --max-time $INTERVAL --connect-timeout $INTERVAL "$JENKINS_MASTER"; do

    ((i++))
    if [[ $i -ge $MAX_RETRIES ]]; then
      echo "ERROR: $(date +%T)[$i]: Failed to connected after $MAX_RETRIES; aborting"
      exit 1
    fi

    echo "INFO: $(date +%T)[$i]: Retry connecting to $JENKINS_MASTER"
    sleep ${WAIT:-1s}
  done

  echo "INFO: $(date +%T): Successfully connected to $JENKINS_MASTER"
}

```

Reply:
> Shouldn't JENKINS_MASTER="http://localhost:8080" be JENKINS_MASTER="http://jenkins:8080" 

obviously :)

Reply:
validateConnection() {
  i=1
  echo "INFO: $(date +%T)[$i]: Checking connection to $JENKINS_URL"
  until curl -L --max-time $INTERVAL --connect-timeout $INTERVAL "$JENKINS_URL"; do

    ((++i))
    if [[ $i -ge $MAX_RETRIES ]]; then
      echo "ERROR: $(date +%T)[$i]: Failed to connected after $MAX_RETRIES; aborting"
      exit 1
    fi

    echo "INFO: $(date +%T)[$i]: Retry connecting to $JENKINS_URL"
    sleep ${INTERVAL:-1s}
  done

  echo "INFO: $(date +%T): Successfully connected $JENKINS_URL"
  return $?
}

MAX_RETRIES=${MAX_RETRIES:-5}
INTERVAL=${INTERVAL:-1}
JENKINS_URL="${JENKINS_URL:-http://jenkins:8080}"

validateConnection

Reply:
```bash
validateConnection() {
  i=1
  echo "INFO: $(date +%T)[$i]: Checking connection to $JENKINS_HOSTNAME"
  # until curl -L --max-time $INTERVAL --connect-timeout $INTERVAL "$JENKINS_HOSTNAME"; do
  until ping -c 1 -W $PING_TIMEOUT "$JENKINS_HOSTNAME"; do

    ((++i))
    if [[ $i -ge $MAX_RETRIES ]]; then
      echo "ERROR: $(date +%T)[$i]: Failed to connected after $MAX_RETRIES; aborting"
      exit 1
    fi

    echo "INFO: $(date +%T)[$i]: Retry connecting to $JENKINS_HOSTNAME"
    sleep $INTERVAL
  done

  echo "INFO: $(date +%T): Successfully connected $JENKINS_HOSTNAME"
  return $?
}

MAX_RETRIES=${MAX_RETRIES:-5}
INTERVAL=${INTERVAL:-1}
JENKINS_HOSTNAME="${JENKINS_HOSTNAME:-jenkins}"
PING_TIMEOUT="${PING_TIMEOUT:-5}"

validateConnection
```